### PR TITLE
[WIP] Adding metadata to sacc file

### PR DESCRIPTION
**Pull request by xgarrido**
The `dict` file is fully stored each field being a string. Other information such as module version or author, date are also added. To be tested and amended

### CODE REVIEW

**Comment by itrharrison** [comment]
Great to see this, I will try and make sure SOLikeT ties up with this metadata properly. A couple of comments for now:

- Would it be better to have separate entries for `['software']` and `['softwareVersion']` as I think they will be easier to parse without the whitespace?
- Can we add a `['paramFile']` or `['inputFile']`? Although I'm not familiar with how the simulations are done, so maybe it is not as simple as there being a single input file.

**Comment by xgarrido** [comment]
> * Would it be better to have separate entries for `['software']` and `['softwareVersion']` as I think they will be easier to parse without the whitespace?

You want a metadata entry with a list of software used by the simulation pipeline +  a metadata entry for each software version ? Something like
```python
software = ["pspy", "pspipe", "mflike"]
cov_sacc.metadata["software"] = str(software)
for s in software:
  cov_sacc.metadata[f"{s} version"] = importlib.import_module(s).__version__
```
Since the `metadata` bank is a `dict`, I don't think it's hard to parse but ok, I can add a `_` between software and version. 

> * Can we add a `['paramFile']` or `['inputFile']`? Although I'm not familiar with how the simulations are done, so maybe it is not as simple as there being a single input file.

The simulation is done *via* a `dict` file something like https://github.com/simonsobs/PSpipe/blob/master/project/maps2params/paramfiles/global_car.dict From the metadata, we can recreate the one used by the simulation by just dumping it within a file. But I agree, even if we store the `dict` file inside the metadata bank, it will be also good to have the original file somewhere and to save its place. We need to be more "professional" in the process of simulation release.
